controllers/renderphase_controller.go

```diff
@@ -312,56 +320,35 @@ func (r *HelmChartInstallReconciler) generateConfigMap(name string, fileName str
 	return cm, nil
 }
 
-// This is in its own method because it will involve
-// reading through multiple levels of the configuration to see
-// what the engine type is. It may even involve inferring an engine
-// type based on the stack contents (though that may happen earlier
-// in the lifecycle than this).
-func (r *HelmChartInstallReconciler) getEngineType(
-	claim *unstructured.Unstructured,
-	sc *v1alpha1.StackConfiguration,
-) string {
-	return "helm2"
-}
-
-func (r *HelmChartInstallReconciler) executeBehavior(
-	ctx context.Context,
-	claim *unstructured.Unstructured,
-	engineCfg *corev1.ConfigMap,
-	targetStackImage string,
-	behavior *v1alpha1.StackConfigurationBehavior,
-) (*unstructured.Unstructured, error) {
-	for _, resource := range behavior.Resources {
-		// TODO error handling
-		// TODO use result
-		r.executeHook(ctx, claim, engineCfg, targetStackImage, resource)
-	}
-	// TODO return a real result
-	return &unstructured.Unstructured{}, nil
-}
-
 // TODO we could have a method create the job, and a higher-level one execute it.
-func (r *HelmChartInstallReconciler) executeHook(
+func (r *RenderPhaseReconciler) executeHook(
 	ctx context.Context,
 	claim *unstructured.Unstructured,
 	engineCfg *corev1.ConfigMap,
 	targetStackImage string,
-	targetResourceDir string,
+	hookCfg *v1alpha1.HookConfiguration,
 ) (*unstructured.Unstructured, error) {
 	// TODO if there is no config specified, either use an empty config or don't specify
 	// one at all.
 
+	// TODO if we change this to meta.AsController, and we have the controller-runtime controller configured
+	// to Own Jobs, then we'll get a reconcile call when Jobs finish. However, we'd need to change the logic
+	// for the reconcile a bit to support that effectively. For example:
+	// - We wouldn't want to create jobs every time reconcile is run
+	//   * This means keeping track of created jobs somewhere and could also mean using deterministic job names
```

Comment:
The name would probably include some metadata of the CR that caused job's creation.